core/src/test/java/org/apache/calcite/test/RelBuilderTest.java

```diff
@@ -2317,6 +2318,69 @@ private static RelNode groupIdRel(RelBuilder builder, boolean extra) {
     assertThat(root, hasTree(expected));
   }
 
+  @ParameterizedTest
+  @ValueSource(booleans = {false, true})
+  void testUnionTypeDerivation(boolean all) {
+    final RelBuilder builder = RelBuilder.create(config().build());
```

Comment:
I'm not sure if this is the best place for these tests. I'm open to suggestions about where they should go.

Reply:
Hi @vbarua , I find some test in https://github.com/apache/calcite/blob/main/core/src/test/java/org/apache/calcite/test/JdbcTest.java#L4525-L4526 . Looks it is a good please for this tests.